### PR TITLE
#2032 - Add corresponding matrix type and vector type

### LINKED ISSUE #2032: Add corresponding vector type and matrix type
`Vector` <--> `Matrix`
`SparseVector` <--> `SparseMatrixCSC`
etc., and using regular arrays as fallback
cf. implementation in Polyhedra.jl: https://github.com/JuliaPolyhedra/Polyhedra.jl/blob/master/src/Polyhedra.jl#L48

### PR DESCRIPTION
**Pull request by mforets**
Closes #2032.

### CODE REVIEW

File: src/Arrays/matrix_vector_operations.jl [hunk 1]
```diff
@@ -20,3 +20,57 @@ function inner(x::AbstractVector{N}, A::AbstractMatrix{N}, y::AbstractVector{N}
...
 end
+
+"""
+    _vector_type(T)
+
+Return a corresponding vector type with respect to type `T`.
+
+### Input
+
+- `T` -- vector type
```
Comment (schillic) [inline src/Arrays/matrix_vector_operations.jl]:
```suggestion
- `T` -- matrix type
```

Comment (mforets) [reply src/Arrays/matrix_vector_operations.jl]: Actually, this should be either vector or matrix type.

Comment (mforets) [inline src/Arrays/matrix_vector_operations.jl]:
```suggestion
- `T` -- vector or matrix type
```

File: src/Arrays/matrix_vector_operations.jl [hunk 2]
```diff
@@ -20,3 +20,57 @@ function inner(x::AbstractVector{N}, A::AbstractMatrix{N}, y::AbstractVector{N}
...
+function _vector_type end
+
+"""
+    _matrix_type(T)
+
+Return a corresponding matrix type with respect to type `T`.
+
+### Input
+
+- `T` -- vector type
```
Comment (mforets) [inline src/Arrays/matrix_vector_operations.jl]:
```suggestion
- `T` -- vector or matrix type
```

**Comment by mforets** [comment]
> Maybe add to the documentation that the Otherwise case may change in the future.

one comment here: we do have to add more dispatches eg for `SArray` and `MArrays` types, but the "otherwise" case is still going to be the fallback. Also, these functions can be easily extended by users code, so we have set types and set operations (once we use these functions, were appropriate) that "just work" with such custom vector/array type pairs in a predictable way.

**Pull request merged**